Resolve `android_build_all_packages` Warnings

jesswrd opened the pull request:
Addressed versioning warnings by updating the following dependencies to their minimum supported versions:

Gradle: 8.7
AGP: 8.6.0
KGP: 2.1.0

## Pre-Review Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter].
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or I have commented below to indicate which [version change exemption] this PR falls under[^1].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or I have commented below to indicate which [CHANGELOG exemption] this PR falls under[^1].
- [x] I updated/added any relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or I have commented below to indicate which [test exemption] this PR falls under[^1].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

**Note**: The Flutter team is currently trialing the use of [Gemini Code Assist for GitHub](https://developers.google.com/gemini-code-assist/docs/review-github-code). Comments from the `gemini-code-assist` bot should not be taken as authoritative feedback from the Flutter team. If you find its comments useful you can update your code accordingly, but if you are unsure or disagree with the feedback, please feel free to wait for a Flutter team member's review for guidance on which automated comments should be addressed.

[^1]: Regular contributors who have demonstrated familiarity with the repository guidelines only need to comment if the PR is not auto-exempted by repo tooling.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[the auto-formatter]: https://github.com/flutter/packages/blob/main/script/tool/README.md#format-code
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[version change exemption]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[CHANGELOG exemption]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test exemption]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

stuartmorgan-g reviewed (commented):
Thanks for investigating this! I wonder why it would be flaking rather than deterministically failing if these are the issues though. Do we have inconsistent tooling on bots? In theory that shouldn't be the case, since .ci.yaml should be setting them.

jesswrd commented on .ci/legacy_project/all_packages/android/app/build.gradle:
done

jesswrd commented:
> Thanks for investigating this! I wonder why it would be flaking rather than deterministically failing if these are the issues though. Do we have inconsistent tooling on bots? In theory that shouldn't be the case, since .ci.yaml should be setting them.

@stuartmorgan-g Instead of changing the tooling, I paired wtih Reid to resolve `.DS_Store` issue within my system. I reverted the changes to this command. I had thought this could potentially be the source of flaking because I could not build the app at all locally, but it was due to `.DS_Store` issue. The timeout flake we are seeing is something else entirely. Still looking into that. 

In the meantime, I updated this PR to only address the versioning warnings. That is still a necessary change.

stuartmorgan-g commented on .ci/legacy_project/README.md:
```diff
@@ -51,3 +51,13 @@ and then deleting everything but `android/` from it:
+  one that we support.
+- Modifies `settings.gradle` to upgrade the Android Gradle Plugin (AGP)
+  from version 8.3.0 to 8.6.0. If a user runs into an error with the AGP
+  version, the warning is clear on how to upgrade the version to one that we support.
```
These should be folded into the bullet points above, just updating the final version. E.g., "... upgrade the Gradle version from 6.7 to 8.7. ...". The goal here isn't to have a history of the specific incremental changes, but a catalog of what is different between a legacy app and the current state.

jesswrd replied:
done

stuartmorgan-g reviewed (approved):
LGTM with one nit.

> Instead of changing the tooling, I paired wtih Reid to resolve my local`.DS_Store` issue . I reverted the changes to this command. I had thought this could potentially be the source of flaking because I could not build the app locally, but it was due to `.DS_Store` issue.

Please feel free to send me a PR that has just those tool changes; there's no reason not to have the tooling be robust against local automatic `.DS_Store` creation to spare future developers the debugging you went through.
